addons/web/static/src/views/pivot/pivot_view.scss

```diff
@@ -21,79 +21,42 @@
             padding: 0!important;
         }
         background-color: $o-view-background-color;
-        width: auto;  // bootstrap override
 
         // Inform the user that he may click on a cell to be redirected to a list view of the
         // items corresponding to the clicked cell
         &.o_enable_linking {
             .o_pivot_cell_value:not(.o_empty):hover {
                 color: $o-brand-primary;
                 cursor: pointer;
-
             }
         }
 
-        thead th:not(.o_pivot_header_cell_closed):not(.o_pivot_header_cell_opened):not(.o_pivot_header_cell) {
-            text-align: center;
-        }
-
         th, td {
-            border-color: gray('200');
-            border-width: $border-width $border-width $border-width $border-width;
+            border-color: gray('300');
+            border-width: $border-width;
         }
 
         th {
             font-weight: $font-weight-normal;  // bootstrap override
             background-color: gray('100');
         }
 
-        @mixin o-pivot-header-cell {
+        .o_pivot_measure_row,
+        .o_pivot_origin_row,
+        .o_pivot_header_cell_closed,
+        .o_pivot_header_cell_opened {
             cursor: pointer;
-            white-space: nowrap;
             user-select: none;
             &:hover {
                 background-color: gray('200');
                 color: $headings-color;
```

Comment:
For the color-on-hover variation you can use the mixin https://github.com/odoo/odoo/blob/master/addons/web/static/src/legacy/scss/utils.scss#L182